### ISSUE TITLE
Crashing

### ISSUE THREAD

**Issue by salling** [comScore/Comscore-Swift-Package-Manager#6]
We've been seeing a lot of crashers from the ComScore framework (for as long as I can remember), but figured I should take this opportunity to try and help.

The biggest group of crashers appear to happen when the app is going to be terminated in the background:

While main thread is doing this:
```
libsystem_c.dylib
exit
UIKitCore
-[UIApplication _terminateWithStatus:]
UIKitCore
-[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:]
UIKitCore
-[_UISceneLifecycleMultiplexer forceExitWithTransitionContext:scene:]
UIKitCore
-[UIApplication workspaceShouldExit:withTransitionContext:]
```

ComScore crashes like this (always the same offsets; sorry no symbols):
```
libsystem_platform.dylib
_sigtramp
libsystem_pthread.dylib
pthread_kill
libsystem_c.dylib
abort
libsystem_malloc.dylib
malloc_vreport
libsystem_malloc.dylib
malloc_report
libsystem_malloc.dylib
free
ComScore
0x102b54000 + 437332
ComScore
0x102b54000 + 436912
ComScore
0x102b54000 + 60460
ComScore
0x102b54000 + 113232
ComScore
0x102b54000 + 107696
ComScore
0x102b54000 + 101772
ComScore
0x102b54000 + 100952
ComScore
0x102b54000 + 351312
ComScore
0x102b54000 + 356280
ComScore
0x102b54000 + 81436
ComScore
0x102b54000 + 323452
ComScore
0x102b54000 + 591244
ComScore
0x102b54000 + 607044
libsystem_pthread.dylib
_pthread_start
libsystem_pthread.dylib
thread_start
```

Xcode organiser seems to ignore this type of termination crashers, which is a shame b/c other ComScore crashers in Xcode organiser are symbolicated (not on AppCenter).

Smaller crash group (but still top crashers in our app) looks like so:

```
Exception Type:  EXC_BAD_ACCESS (SIGSEGV)
Exception Subtype: KERN_INVALID_ADDRESS at 0x0000f4e674526680 -> 0x0000006674526680 (possible pointer authentication failure)
VM Region Info: 0x6674526680 is in 0x1000000000-0x7000000000;  bytes after start: 371318744704  bytes before end: 40998115711
      REGION TYPE                 START - END      [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      commpage (reserved)      fc0000000-1000000000 [  1.0G] ---/--- SM=NUL  ...(unallocated)
--->  GPU Carveout (reserved) 1000000000-7000000000 [384.0G] ---/--- SM=NUL  ...(unallocated)
      UNUSED SPACE AT END

Termination Signal: Segmentation fault: 11
Termination Reason: Namespace SIGNAL, Code 0xb
Terminating Process: exc handler [79574]
Triggered by Thread:  3

Thread 3 name:
Thread 3 Crashed:
0   ComScore                      	0x0000000104f86f70 ComScore::ContextTasksThread<ComScore::Core>::run() + 144
1   ComScore                      	0x0000000104f86f58 ComScore::ContextTasksThread<ComScore::Core>::run() + 120
2   ComScore                      	0x0000000104fc858c ComScore::Thread::threadEntryPoint() + 104
3   ComScore                      	0x0000000104fcc344 threadEntryProc + 32
4   libsystem_pthread.dylib       	0x00000001f363fcb0 _pthread_start + 320 (pthread.c:881)
5   libsystem_pthread.dylib       	0x00000001f3648778 thread_start + 8
```

Thanks!

**Comment by markoocko** [comment]
Hi Jonas,
Thank you for the detailed explanation. We are acually currently investigating a similar issue. Would you be able to send us the original crash report and also let us know which version of our library this crash was produced on? That would help us a lot. 
Thanks for your help.

**Comment by markoocko** [comment]
Newly released v6.8.1 should fix the above issue. Please let us know if you still see these crashes after the update.
